flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -252,6 +255,22 @@ public void setFailOnCheckpointingErrors(boolean failOnCheckpointingErrors) {
 		this.failOnCheckpointingErrors = failOnCheckpointingErrors;
 	}
 
+	/**
+	 * Get the tolerable checkpoint failure number which used by the checkpoint failure manager
+	 * to determine when we need to fail the job.
+	 */
+	public int getTolerableCheckpointFailureNumber() {
+		return tolerableCheckpointFailureNumber;
+	}
+
+	/**
+	 * Set the tolerable checkpoint failure number, the default value is 0 that means
+	 * we do not tolerance any checkpoint failure.
+	 */
+	public void setTolerableCheckpointFailureNumber(int tolerableCheckpointFailureNumber) {
```

Comment:
I noticed that you have introduced variable `tolerableCheckpointFailureNumber` and related setter here. However, I cannot see any method to call this method. Why not introduce related configuration in `CheckpointingOptions` so that user could configure this variable?

Reply:
IMO, you are wrong. `CheckpointingOptions` defines some global config options which almost are related to the **state**. But `tolerableCheckpointFailureNumber ` a checkpoint specific config option belongs per job and it will enhance the config option of `setFailOnCheckpointingErrors`. It should not be defined in `CheckpointingOptions`. Currently, is it not used? Yes. Because we split this feature into three steps. It will be used in the third step. But in this step, we need it to create the `CheckpointFailureManager`. I think you know the reason now.